src/Hooks/ScheduledTaskListener.php

```diff
@@ -49,4 +44,13 @@ public function __invoke(ScheduledTaskFinished|ScheduledTaskSkipped|ScheduledTas
 
         $this->nightwatch->digest();
     }
+
+    private function isFinishedEventForFailedTask(ScheduledTaskFinished|ScheduledTaskSkipped|ScheduledTaskFailed $event): bool
+    {
+        return Compatibility::$firesFinishedAndFailedEventsForScheduledConsoleCommands &&
+            $event instanceof ScheduledTaskFinished &&
+            $event->task->command !== null &&
+            $event->task->exitCode !== 0 &&
+            ! $event->task->runInBackground;
```

Comment:
Scheduled tasks that run in the background don't return an exit code (i.e. null).